backend/Application/Aggregates/Contract/ContractRepository.cs

```diff
@@ -155,6 +162,36 @@ public async Task<long> GetReadOnlyLatestImportState(CancellationToken token)
             .FirstOrDefaultAsync(token);
     }
     
+    /// <summary>
+    /// Looking after <see cref="ContractInitialized"/> event for <see cref="contractAddress"/>. First the event is searched
+    /// for in the change provider of Entity Framework. These are the entity which has been added in the current transaction
+    /// but are not yet committed to the database.
+    ///
+    /// If <see cref="ContractInitialized"/> has been added in this transaction the database it queried for the event.
+    /// </summary>
+    public async Task<ContractInitialized> GetReadonlyContractInitializedEventAsync(ContractAddress contractAddress)
+    {
+        var contractInitialized = _context.ChangeTracker
+            .Entries<ContractEvent>()
+            .Select(e => e.Entity)
+            .Where(e => e.ContractAddressIndex == contractAddress.Index && e.ContractAddressSubIndex == contractAddress.SubIndex)
+            .Where(e => e.Event is ContractInitialized)
+            .Select(e => (e.Event as ContractInitialized)!)
+            .FirstOrDefault();
+
+        if (contractInitialized != null)
+        {
+            return contractInitialized;
+        }
+
+        var connection = _context.Database.GetDbConnection();
+        var parameter = new { Index = (long)contractAddress.Index, Subindex = (long)contractAddress.SubIndex};
+        var contractEvents = (await connection.QueryAsync<ContractEvent>(ContractEvent.ContractInitializedEventSql, parameter))
```

Comment:
If the event doesn’t exist an exception is thrown. I think this is good behaviour since one is failing as fast a possible.

I have added a comment to the method docs about this.